src/iron.rs

```diff
@@ -111,89 +128,139 @@ impl<H: Handler> Iron<H> {
     /// Kick off the server process using the HTTP protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn http<A>(self, addr: A) -> HttpResult<Listening>
+    pub fn http<A>(self, addr: A) -> HttpResult<()>
         where A: ToSocketAddrs
     {
-        HttpListener::new(addr).and_then(|l| self.listen(l, Protocol::http()))
+        let addr = addr.to_socket_addrs()?.next().unwrap();
+
+        let core = Core::new().unwrap();
+        let handle = core.handle();
+
+        let sock = TcpListener::bind(&addr, &handle).unwrap().incoming();
+
+        return self.listen(sock, addr, Protocol::http(), core, handle);
     }
 
     /// Kick off the server process using the HTTPS protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn https<A, S>(self, addr: A, ssl: S) -> HttpResult<Listening>
-        where A: ToSocketAddrs,
-              S: 'static + SslServer + Send + Clone
+    #[cfg(feature = "ssl")]
```

Comment:
For Servers, the idea is that `hyper::server::Http` is a `ServerProto`, and can be wrapped in others. So, it doesn't care where the bytes are coming through, you can pipe them in from anything. An example of doing this using tokio-tls is:

```rust
let http = Http::new();
let tls = tokio_tls::proto::Server::new(http, native_tls_acceptor);
let tcp = tokio_proto::TcpServer::new(tls, addr);
// ...
```